### PR TITLE
Feature/optimization

### PR DESCRIPTION
**Pull request by LouisRouss**
This PR fixes a certain number of issues coming with the torch compile optimization

### CODE REVIEW

File: src/diffulab/training/trainer.py [hunk 1]
```diff
@@ -407,8 +375,10 @@ def train(
                     if isinstance(v, torch.Tensor):
                         state[k] = v.to(device)
 
-        for loss_idx in range(len(diffuser.extra_losses)):
-            diffuser.extra_losses[loss_idx] = self.accelerator.prepare(diffuser.extra_losses[loss_idx])  # type: ignore
+        additional_prepared: "list[nn.Module | DistributedDataParallel | FullyShardedDataParallel]" = []
+        for loss in diffuser.extra_losses:
+            loss.set_model(diffuser.denoiser)  # type: ignore
+            additional_prepared.extend(loss.accelerate_prepare(self.accelerator))
```
[the remark this answers is not in the document]
Comment (LouisRouss) [reply src/diffulab/training/trainer.py:382]: unrelated the accelerate prepare of the loss is relative to the different projector that the loss might include
